src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3991,176 +4018,172 @@ protected BoundExpression BindObjectCreationExpression(ObjectCreationExpressionS
 
         private BoundExpression BindDelegateCreationExpression(ObjectCreationExpressionSyntax node, NamedTypeSymbol type, DiagnosticBag diagnostics)
         {
-            // Get the bound arguments and the argument names.
             AnalyzedArguments analyzedArguments = AnalyzedArguments.GetInstance();
+            BindArgumentsAndNames(node.ArgumentList, diagnostics, analyzedArguments, isDelegateCreation: true);
+            var result = BindDelegateCreationExpression(node, type, analyzedArguments, node.Initializer, diagnostics);
+            analyzedArguments.Free();
+            return result;
+        }
 
-            try
+        private BoundExpression BindDelegateCreationExpression(SyntaxNode node, NamedTypeSymbol type, AnalyzedArguments analyzedArguments, InitializerExpressionSyntax initializerOpt, DiagnosticBag diagnostics)
```

Comment:
:memo: note to reviewers: github diff is getting largely confused here. This file is easier to review in CodeFlow.